interactive_engine/executor/ir/graph_proxy/src/apis/graph/element/path.rs

```diff
@@ -44,48 +47,84 @@ impl From<Edge> for VertexOrEdge {
     }
 }
 
-#[derive(Clone, Debug, Hash)]
+#[derive(Clone, Debug)]
 pub enum GraphPath {
-    WHOLE((Vec<VertexOrEdge>, usize)),
-    END((VertexOrEdge, usize)),
+    AllV(Vec<VertexOrEdge>),
+    SimpleAllV((Vec<VertexOrEdge>, HashSet<ID>)),
```

Comment:
Do not think it is a good idea to store `ID` in a hashset, which is very costly and memory-inefficient. Considering that the path often contains less than 4 items, simple iterate the vector to `dedup()` is not very unacceptable. Same for `SimpleEndV`